environment/src/main/java/com/sequenceiq/environment/environment/validation/validators/NetworkCreationValidator.java

```diff
@@ -32,39 +34,80 @@ public NetworkCreationValidator(NetworkService networkService) {
 
     public ValidationResultBuilder validateNetworkCreation(Environment environment, NetworkDto network, Map<String, CloudSubnet> subnetMetas) {
         ValidationResultBuilder resultBuilder = new ValidationResultBuilder();
-        if (AWS.name().equalsIgnoreCase(environment.getCloudPlatform()) && network != null && network.getNetworkCidr() == null) {
-            String message;
-            if (network.getSubnetIds().size() < 2) {
-                message = "Cannot create environment, there should be at least two subnets in the network";
-                LOGGER.info(message);
-                resultBuilder.error(message);
+        if (network != null && network.getNetworkCidr() == null) {
+            if (AWS.name().equalsIgnoreCase(environment.getCloudPlatform())) {
+                validateAwsExistingNetwork(network, subnetMetas, resultBuilder);
+            } else if (AZURE.name().equalsIgnoreCase(environment.getCloudPlatform())) {
+                validateAzureExistingNetwork(network, resultBuilder);
             }
-            if (subnetMetas.size() != network.getSubnetIds().size()) {
-                message = String.format("Subnets of the environment (%s) are not found in the vpc (%s). ",
-                        String.join(",", getSubnetDiff(network.getSubnetIds(), subnetMetas.keySet())),
-                        networkService.getAwsVpcId(network).orElse(""));
+        }
+        validateNetworkIdAndCidr(environment, network, resultBuilder);
+        return resultBuilder;
+    }
+
+    private void validateAwsExistingNetwork(NetworkDto network, Map<String, CloudSubnet> subnetMetas, ValidationResultBuilder resultBuilder) {
```

Comment:
This AWS specific validation should be in AwsEnvironmentNetworkValidator

Reply:
That validator currently is called from the flow only. The task was to validate during request time.
I can of course use this validator from the EnvironmentCreationService as well, if this is what you meant.

Reply:
Yes, that's what I meant